tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/onsi/gomega"
+	"github.com/vmware/govmomi/find"
+	vsan "github.com/vmware/govmomi/vsan"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+)
+
+type FaultDomains struct {
+	primarySiteHosts   []string
+	secondarySiteHosts []string
+	witness            string
+	hostsDown          []string `default:"[]"`
+	// isNetworkPartitioned bool     `default:"false"`
+}
+
+var fds FaultDomains
+
+//initialiseFdsVar initialise fds variable
+func initialiseFdsVar(ctx context.Context) {
+	fdMap := createFaultDomainMap(ctx, &e2eVSphere)
+	hostsWithoutFD := []string{}
+	for host, site := range fdMap {
+		if strings.Contains(site, "rimary") {
+			fds.primarySiteHosts = append(fds.primarySiteHosts, host)
+		} else if strings.Contains(site, "econdary") {
+			fds.secondarySiteHosts = append(fds.secondarySiteHosts, host)
+		} else {
+			hostsWithoutFD = append(hostsWithoutFD, host)
+		}
+	}
+
+	// assuming we don't have hosts which are not part of the vsan stretched cluster in the testbed here
+	gomega.Expect(len(hostsWithoutFD) == 1).To(gomega.BeTrue())
+	fds.witness = hostsWithoutFD[0]
+
+}
+
+//siteFailover cause a site failover by powering off hosts of the given site
+func siteFailover(primarySite bool) {
+	hostsToPowerOff := fds.secondarySiteHosts
+	if primarySite {
+		hostsToPowerOff = fds.primarySiteHosts
+	}
+	framework.Logf("hosts to power off: %v", hostsToPowerOff)
+	powerOffHostParallel(hostsToPowerOff)
+}
+
+//powerOffHostParallel powers off given hosts
+func powerOffHostParallel(hostsToPowerOff []string) {
+	hostlist := ""
+	for _, host := range hostsToPowerOff {
+		for _, esxHost := range tbinfo.esxHosts {
+			if esxHost["ip"] == host {
+				hostlist += esxHost["vmName"] + " "
+			}
+		}
+		fds.hostsDown = append(fds.hostsDown, host)
+	}
+
+	err := vMPowerMgmt(tbinfo.user, tbinfo.location, hostlist, false)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	for _, host := range hostsToPowerOff {
+		err = waitForHostToBeDown(host)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
+}
+
+//siteRestore restores a site by powering on hosts of the given site
```

Comment:
// <function_name>

Reply:
didn't get what you were looking for here